core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -169,23 +279,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         for (GeoPoint point : points) {
             builder.value(point);
         }
-        for (String geohash : geohashes) {
-            builder.value(geohash);
-        }
         builder.endArray();
 
-        if (unit != null) {
-            builder.field("unit", unit);
-        }
-        if (geoDistance != null) {
-            builder.field("distance_type", geoDistance.name().toLowerCase(Locale.ROOT));
-        }
+        builder.field("unit", unit);
+        builder.field("distance_type", geoDistance.name().toLowerCase(Locale.ROOT));
         if (order == SortOrder.DESC) {
             builder.field("reverse", true);
         }
-        if (sortMode != null) {
-            builder.field("mode", sortMode);
-        }
+        builder.field("mode", sortMode);
```

Comment:
See comment above, if sortMode can be `null`, this would need to be wrapped in check again.